store/rootmulti/store.go

```diff
@@ -77,6 +77,8 @@ type Store struct {
 	listeners map[types.StoreKey]*types.MemoryListener
 
 	metrics metrics.StoreMetrics
+
+	mtx sync.Mutex
```

Comment:
What is the purpose of this mutex? What is is meant to guard? At the moment you lock it in PopStateCache, to read rs.listeners, but in that case you also need to lock it in any other method that accesses rs.listeners, for example AddListeners.

Reply:
updated to lock calls to rs.listeners.